cdap-ui/app/hydrator/controllers/list-ctrl.js

```diff
@@ -130,87 +188,133 @@ angular.module(PKG.name + '.feature.hydrator')
       api.runs(params)
         .$promise
         .then(function (runs) {
-          app._stats.numRuns = runs.length;
-          app._stats.lastStartTime = runs.length > 0 ? runs[0].start : 'N/A';
+          if (runs.length) {
+            app._stats.numRuns = runs.length;
+            app._stats.lastStartTime = runs.length > 0 ? runs[0].start : 'N/A';
+            var currentRun = runs[0];
+            setDurationTimers(app, currentRun);
+            app._latest = currentRun;
+            statusMap[app.id] = vm.MyPipelineStatusMapper.lookupDisplayStatus(app._latest.status);
+          } else {
+            statusMap[app.id] = vm.MyPipelineStatusMapper.lookupDisplayStatus('SUSPENDED');
+          }
+          updateStatusAppObject();
+        });
+    }
 
-          for (var i = 0; i < runs.length; i++) {
-            var status = runs[i].status;
+    function getPipelines() {
+      myPipelineApi.list({
+        namespace: $stateParams.namespace
+      })
+        .$promise
+        .then(function success(res) {
+          vm.pipelineList = res;
 
-            if (['RUNNING', 'STARTING', 'STOPPING'].indexOf(status) === -1) {
-              app._latest = runs[i];
-              break;
-            }
-          }
+          fetchDrafts().then(setCurrentPage);
+
+          angular.forEach(vm.pipelineList, function (app) {
+            app._stats = {};
+
+            fetchRunsInfo(app);
+          });
 
+          fetchStatus();
+          fetchWorkflowNextRunTimes();
         });
     }
 
+    /**
+     * Gets the next workflow run times. This must be called after batch objects have been created.
+     */
+    function fetchWorkflowNextRunTimes() {
+      batch.forEach(function (batchParams) {
+        dataSrc.request({
+          _cdapNsPath: '/apps/' + batchParams.appId +'/workflows/' + batchParams.programId +'/nextruntime',
+        })
+        .then(function (res) {
+          if (res && res.length) {
+            vm.pipelineList.forEach(function (app) {
+              if (app.id === batchParams.appId) {
+                app._stats.nextRun = res[0].time;
+              }
+            });
+          }
+        });
+      });
+    }
+
+    function setDurationTimers(app, run) {
+      if (run.status === 'RUNNING') {
+
+        if (!app.pipelineDurationTimer) {
+          app.pipelineDurationTimer = vm.$interval(() => {
```

Comment:
Since we are not polling for the pipeline status, this duration will continue on forever...